lib/index.ts

```diff
@@ -119,10 +132,24 @@ export function fromImageDescriptors(
 			reject(new TarError(e));
 		});
 
-		buildStream.pipe(extract);
+		newStream.pipe(extract);
 	});
 }
 
+export function buildHasSecrets(tasks: BuildTask[]): boolean {
+	if (tasks.length === 0) {
+		return false;
+	}
+
+	return !_.isEmpty(
+		generateSecretPopulationMap(
+			_.map(tasks, 'serviceName'),
+			tasks[0].buildMetadata,
+			'/tmp',
```

Comment:
Maybe `os.tmpdir()` would be better ?

Reply:
For this it doesn't make a difference, we're just seeing if there would be any secrets to populate. My thoughts for this are that because it won't work on a regular docker daemon, we should surface a way for the caller to detect if build secrets are being used and report that this will only work with balena-engine.